broker/providers/ansible_tower.py

```diff
@@ -22,13 +25,65 @@
 from broker import helpers
 from broker.providers import Provider
 
-# ruamel has a hard time with PseudoNamespace objects
+# Configure ruamel yaml representer for PseudoNamespace
 yaml.representer.add_representer(
     awxkit.utils.PseudoNamespace,
     lambda dumper, data: dumper.represent_dict(dict(data)),
 )
 
 
+def detect_and_reconfigure_for_aap25(base_url):
+    """Detect AAP 2.5+ and reconfigure awxkit if needed.
+
+    This function can be called multiple times safely.
+    Returns True if reconfiguration was done, False otherwise.
+    """
+    # Exit immediately if we've already configured
+    if getattr(detect_and_reconfigure_for_aap25, "_configured", False):
+        return False
+
+    # Check explicit setting configuration first
+    if not (need_reconfigure := settings.ANSIBLETOWER.get("AAP_VERSION") == "2.5"):
```

Comment:
```suggestion
    if not (need_reconfigure := str(settings.ANSIBLETOWER.get("AAP_VERSION")) == "2.5"):
```
I'm hitting an issue with type conversion. If I set `aap_version: "2.5"` in the config file, despite having the number quoted, it's still handled as a float, not as a string. This causes the comparions with "2.5" string to be false. Either explicitly converting the version from config to string, or having ` == 2.5` without quotes fixes the issue. 

Besides, if the setting is present as `aap_version: ` (no version string defined) as produced by current config migration, the conditional check assumes that the AAP is not 2.5, and the API check below is never performed, resulting in a failure. The version check should be more explicit (check for null, check for the Y version?).
Maybe dropping that `return False` would be better, to always run the autodetection (unless explicit "2.5", which goes for reconfig right away? Just an idea.